packages/server/lib/controllers/auth/postTwoStep.ts

```diff
@@ -0,0 +1,204 @@
+
+        await hmacCheck({
+            environment,
+            logCtx,
+            providerConfigKey,
+            connectionId: receivedConnectionId,
+            hmac,
+            res
+        });
+
+        const config = await configService.getProviderConfig(providerConfigKey, environment.id);
+        if (!config) {
+            await logCtx.error('Unknown provider config');
+            await logCtx.failed();
+            res.status(404).send({ error: { code: 'unknown_provider_config' } });
+            return;
+        }
+
+        const provider = getProvider(config.provider);
+        if (!provider) {
+            await logCtx.error('Unknown provider');
+            await logCtx.failed();
+            res.status(404).send({ error: { code: 'unknown_provider_template' } });
+            return;
+        }
+
+        if (provider.auth_mode !== 'TWOSTEP') {
+            await logCtx.error('Provider does not support TWOSTEP auth', { provider: config.provider });
+            await logCtx.failed();
+            res.status(400).send({ error: { code: 'invalid_auth_mode' } });
+            return;
+        }
+
+        await logCtx.enrichOperation({ integrationId: config.id!, integrationName: config.unique_key, providerName: config.provider });
+
+        const {
+            success,
+            error,
+            response: credentials
+        } = await connectionService.getTwoStepCredentials(provider as ProviderTwoStep, bodyData, connectionConfig);
+
+        if (!success || !credentials) {
+            await logCtx.error('Error during TwoStep credentials creation', { error, provider: config.provider });
+            await logCtx.failed();
+
+            errorManager.errRes(res, 'perimeter_error');
+
+            return;
+        }
+
+        const connectionId = receivedConnectionId || connectionService.generateConnectionId();
+
+        await logCtx.info('TwoStep connection creation was successful');
+        await logCtx.success();
+
+        const [updatedConnection] = await connectionService.upsertPerimeterConnection({
+            connectionId,
+            providerConfigKey,
+            credentials,
+            connectionConfig,
+            metadata: {},
+            config,
+            environment,
+            account
+        });
+
+        if (updatedConnection) {
+            await logCtx.enrichOperation({ connectionId: updatedConnection.connection.id!, connectionName: updatedConnection.connection.connection_id });
+            void connectionCreatedHook(
+                {
+                    connection: updatedConnection.connection,
+                    environment,
+                    account,
+                    auth_mode: 'TWOSTEP',
+                    operation: updatedConnection.operation
+                },
+                config.provider,
+                logContextGetter,
+                undefined,
+                logCtx
+            );
+        }
```

Comment:
Sorry to inflict you that, I just changed this block of code in all auth mode can you apply the same changes please? ☺️ 
https://github.com/NangoHQ/nango/pull/2884/files

Reply:
No worries at all—I initially missed this. Thank you! ☺️ I've updated it with the latest [commit](https://github.com/NangoHQ/nango/pull/2868/commits/05b56ec0584bccafc644ef6191387894f4a39512)